src/compiler/core.ts

```diff
@@ -1011,6 +1006,17 @@ namespace ts {
         return keys;
     }
 
+    export function getOwnValues<T>(sparseArray: T[]): T[] {
+        const values: T[] = [];
+        for (const key in sparseArray) {

```

Comment:
I'm sure I'm missing something, but how is this different from a for-of loop?

Reply:
A `for-in` loop iterates over the keys of an object, while a `for-of` loop only works on an object implementing the iterator protocol. See [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators)

Reply:
It's also supposedly identical to `Object.values`, which, much like `hasOwnProperty`, we should probably just check for the existence of and use if available.